src/components/clusterDetail/ClusterInstallationProgressCard.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { Card, Title, CardBody, CardHeader, CardExpandableContent } from '@patternfly/react-core';
+import { Cluster, Credentials, getClusterCredentials, stringToJSON } from '../../api';
+import { AlertsContextProvider } from '../AlertsContextProvider';
+import CancelInstallationModal from './CancelInstallationModal';
+import ClusterCredentials from './ClusterCredentials';
+import ClusterInstallationError from './ClusterInstallationError';
+import ClusterProgress from './ClusterProgress';
+import FailedHostsWarning from './FailedHostsWarning';
+import ResetClusterModal from './ResetClusterModal';
+import HostsTable from '../hosts/HostsTable';
+import ClusterDetailsButtonGroup from './ClusterDetailsButtonGroup';
+import { ClusterStatusIcon } from '../clusters/ClusterStatus';
+
+import './ClusterInstallationProgressCard.css';
+
+type AIClusterDetailsCardProps = {
+  clusterObj: Cluster;
+};
+
+const getID = (suffix: string) => `cluster-detail-${suffix}`;
+
+const AIClusterDetailsCard: React.FC<AIClusterDetailsCardProps> = ({ clusterObj }) => {
+  // CamelCase the cluster object
+  const cluster: Cluster = stringToJSON<Cluster>(JSON.stringify(clusterObj)) || clusterObj;
+
+  const [cancelInstallationModalOpen, setCancelInstallationModalOpen] = React.useState(false);
+  const [resetClusterModalOpen, setResetClusterModalOpen] = React.useState(false);
+  const [credentials, setCredentials] = React.useState<Credentials>();
+  const [credentialsError, setCredentialsError] = React.useState();
+  const [isCardExpanded, setIsCardExpanded] = React.useState(cluster.status !== 'installed');
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError(undefined);
+      try {
+        const response = await getClusterCredentials(cluster.id);
+        setCredentials(response.data);
+      } catch (err) {
+        setCredentialsError(err);
+      }
+    };
+    fetch();
+  }, [cluster.id]);
+
+  React.useEffect(() => {
+    if (cluster.status === 'installed') {
+      fetchCredentials();
+    }
+  }, [cluster.status, fetchCredentials]);
+
+  return (
+    <>
+      <AlertsContextProvider>
```

Comment:
The modals inside do not render alerts, so storing them in that context is equal to ignoring them.
The card bellow should render them, so let's move the modals into the card and wrap it by that context.

Or move the context higher in the hierarchy. Anyway, we need to be sure that a potential error is always shown to the user.